src/features/utils/ansibleCfg.ts

```diff
@@ -29,13 +29,13 @@ export async function scanAnsibleCfg(
   rootPath: string | undefined = undefined
 ): Promise<AnsibleVaultConfig | undefined> {
   /*
-   * Reading order:
+   * Reading order (based on the documentation: https://docs.ansible.com/ansible/latest/reference_appendices/config.html#ansible-configuration-settings):
    * 1) ANSIBLE_CONFIG
    * 2) ansible.cfg (in current workspace)
    * 3) ~/.ansible.cfg
-   * 4) /etc/ansible.cfg
+   * 4) /etc/ansible/ansible.cfg
```

Comment:
We have people keeping ansible.cfg and ansible files in subfolders and this approach does not help them. Any idea what we can do for them? (not as part of tris PR)

Reply:
A similar thing (not the same) has been implemented in the ALS to locate the `.ansible-lint` config file [(ref)](https://github.com/ansible/ansible-language-server/blob/main/src/services/ansibleLint.ts#L236-L263). That is, even if the user has no .ansible-lint file in the CWD, we traverse to the root of the workspace and try to find out config files in any of the parent directories up till the root.

This does solve the issue to some extent.
But, again, if ansible doesn't recognize the config file associated with a playbook(if it does, `ansible --version` should show it), there is no point in writing code to locate it.